app/src/lib/api.ts

```diff
@@ -500,31 +525,27 @@ export class API {
    * pages when available, buffers all items and returns them in
    * one array when done.
    */
-  private async fetchAll<T>(path: string): Promise<ReadonlyArray<T>> {
+  private async fetchAll<T>(path: string, options?: IFetchAllOptions<T>) {
     const buf = new Array<T>()
+    const opts: IFetchAllOptions<T> = { perPage: 100, ...options }
+    const params = { per_page: `${opts.perPage}` }
+    const { NotFound, NotModified } = HttpStatusCode
 
-    const params = {
-      per_page: '100',
-    }
     let nextPath: string | null = urlWithQueryString(path, params)
-
     do {
       const response = await this.request('GET', nextPath)
-      if (response.status === HttpStatusCode.NotFound) {
-        log.warn(`fetchAll: '${path}' returned a 404`)
-        return []
-      }
-      if (response.status === HttpStatusCode.NotModified) {
-        log.warn(`fetchAll: '${path}' returned a 304`)
+      if (response.status === NotFound || response.status === NotModified) {
```

Comment:
While you're in here I wanted to bring to your attention #7019 which was a case where we had a `HTTP 502` that wasn't handled correctly and affected cloning. Any interested in generalizing this to look for ranges of error codes (`3xx`, `4xx`, `5xx`) rather than just a few known error codes?

Reply:
Are you saying that you want to ignore all non-200 status codes?

Reply:
"Ignore" isn't how I'd exactly put it. I had a play around with this in #7020 and had these things in mind at the time:

 - look for known ranges of problem error codes, rather than explicit values
 - log them out as we currently do, so we can identify them later if users spot issues
 - as soon as we encounter an error code, return the accumulated pages of items in `buf` - the app currently returns an empty array, which impacted #7019 because it would occur intermittently in the middle of a bunch of network requests

Reply:
Alright, take a look now and see if that's what you're looking for. Just want to make sure I understand.